fixes https://github.com/mkdocs/mkdocs/issues/728

justinkinney opened the pull request:
Just a possible solution.

waylan commented on mkdocs/utils/__init__.py:
```diff
@@ -431,7 +435,10 @@ def nest_paths(paths):
         for part in parts:
-            part = filename_to_title(part)
+            dir = False
+            if os.path.basename(part) != path:
```
Shouldn't this line read:

```python
if os.path.basename(path) != part
```

waylan commented on mkdocs/utils/__init__.py:
```diff
@@ -431,7 +435,10 @@ def nest_paths(paths):
-            part = filename_to_title(part)
+            dir = False
+            if os.path.basename(part) != path:
+                dir = True
```
Why not change the line above from `!=` to `==` and then change this line to:

```python
part = os.path.splitext(part)[0]
```

Then you can remove the call to `os.path.splitext` in `filename_to_title` and the new `dir` parameter becomes unnecessary.

waylan replied:
Oh right, you can't modify `part` here because the later call to `find_or_create_node` needs the full filename. The general idea should still be doable though.

justinkinney replied:
I'm not sure that works since filename_to_title is called later from `nav.py` to add files to the TOC when `_path_to_page` is called. I tried a few variations on what you described, but to be honest I think they just complicate things a bit more (at least the ways I tried).

waylan replied:
Ah, okay. I wasn't sure if the function was called from anywhere else in the code. Then we definitely need two separate code paths, one for filenames and one for dirnames.

justinkinney commented:
Ok, based on previous comments, I've added a `dirname_to_title()` method to handle the code path in `nest_paths()`. I also simplified my changes since I realized that `nest_paths()` is only working on paths not filenames due to `directory, _ = os.path.split(path)`.

d0ugal commented:
Works great, thanks! Sorry for the long delay.
